monasca_events_api/tests/first_test.py

```diff
@@ -12,10 +12,228 @@
 # under the License.
 
+
+import falcon
+from monasca_events_api.common.messaging import exceptions
+from monasca_events_api.common.repositories.mysql.events_repository import EventsRepository
+from monasca_events_api.v2.events import Events
+
+import mock
+from monasca_events_api.common.repositories.exceptions import RepositoryException
+
+from oslo_utils import timeutils
+
 import unittest
 
 
+class EventsSubClass(Events):
+    def __init__(self):
+        self._default_authorized_roles = ['user', 'domainuser',
+                                          'domainadmin', 'monasca-user']
+        self._post_events_authorized_roles = ['user', 'domainuser',
+                                              'domainadmin', 'monasca-user', 'monasca-agent']
+        self._events_repo = None
+        self._message_queue = None
+        self._region = 'useast'
+
+    def _event_transform(self, event, tenant_id, _region):
+        return dict(event=1, meta=dict(tenantId='0ab1ac0a-2867-402d',
+                    region='useast'), creation_time=timeutils.utcnow_ts())
+
+
 class Test_first(unittest.TestCase):
 
-    def test_first(self):
-        assert 1 == 1
+    @mock.patch('monascaclient.ksclient.KSClient')
+    def _generate_req(self, token):
+        """Generate a mock HTTP request"""
+        req = mock.MagicMock()
+        req.get_param.return_value = None
+
+        req.headers = {
+            'X-Auth-User': 'mini-mon',
+            'X-Auth-Token': token,
+            'X-Auth-Key': 'password',
+            'X-TENANT-ID': '0ab1ac0a-2867-402d',
+            'X-ROLES': 'user, domainuser, domainadmin, monasca-user, monasca-agent',
+            'Accept': 'application/json',
+            'User-Agent': 'python-monascaclient',
+            'Content-Type': 'application/json'}
+
+        req.body = {}
+        req.content_type = 'application/json'
+        return req
+
+    @mock.patch('monasca_events_api.v2.events.Events._list_event')
+    @mock.patch('monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_get_pass(self, helper_tenant_id, helpers_validate, mysqlRepo, listev):
+        """GET Method success"""
+        helpers_validate.validate_authorization.return_value = True
+        listev.return_value = True
+        mysqlRepo.connect.return_value = True
+        helper_tenant_id.get_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        event_id = "0ab1ac0a-2867-402d-83c7-d7087262470c"
+        eventsObj = EventsSubClass()
+        eventsObj._events_repo = EventsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_get(self._generate_req(), res, event_id)
+        except Exception:
+            self.fail("GET() raised HTTPInternalServerError unexpectedly")
```

Comment:
This is a test for failure.  You also need a test for success.  Rather than wrapping with a try just run eventsObj directly and assert that the result body is what you expected.  Right now you have no data coming back, since you stubbed out _list_event, so you need to have _list_event return some sample data and verify that the format of that response is correct after all the on_get processing has been finished.